scripts/cybershake/auto_submit.py

```diff
@@ -307,6 +313,96 @@ def submit_script_to_scheduler(script_name, target_machine=None, **kwargs):
+                        "MGMT_DB_LOC": root_folder,
+                        "REL_NAME": run_name,
+                    }
+                ),
+            ),
+            target_machine=get_target_machine(const.ProcessType.VM_PARAMS).name,
+        )
+    elif proc_type == const.ProcessType.VM_GEN.value:
+        submit_script_to_scheduler(
+            get_platform_specific_script(
+                const.ProcessType.VM_GEN,
+                OrderedDict(
+                    {
+                        "VM_PARAMS_YAML": str(
+                            Path(sim_struct.get_fault_VM_dir(root_folder, run_name))
+                            / "vm_params.yaml"
+                        ),
+                        "OUTPUT_DIR": sim_struct.get_fault_VM_dir(
+                            root_folder, run_name
+                        ),
+                        "SRF_PATH": sim_struct.get_srf_path(root_folder, run_name),
+                        "MGMT_DB_LOC": root_folder,
+                        "REL_NAME": run_name,
+                    }
+                ),
+            ),
+            target_machine=get_target_machine(const.ProcessType.VM_GEN).name,
+        )
+    elif proc_type == const.ProcessType.VM_PERT.value:
+        submit_script_to_scheduler(
+            get_platform_specific_script(
+                const.ProcessType.VM_PERT,
+                OrderedDict(
+                    {
+                        "VM_PARAMS_YAML": str(
+                            Path(sim_struct.get_fault_VM_dir(root_folder, run_name))
+                            / "vm_params.yaml"
+                        ),
+                        "OUTPUT_DIR": sim_struct.get_fault_VM_dir(
+                            root_folder, run_name
+                        ),
+                        "SRF_PATH": sim_struct.get_srf_path(root_folder, run_name),
+                        "MGMT_DB_LOC": root_folder,
+                        "REL_NAME": run_name,
+                    }
+                ),
+            ),
+            target_machine=get_target_machine(const.ProcessType.VM_PERT).name,
+        )
+    elif proc_type == const.ProcessType.INSTALL_FAULT.value:
+        fault_dir = sim_struct.get_fault_dir(
+            root_folder, sim_struct.get_fault_from_realisation(run_name)
+        )
+        submit_script_to_scheduler(
+            get_platform_specific_script(
+                const.ProcessType.INSTALL_FAULT,
+                OrderedDict(
+                    {
+                        "VM_PARAMS_YAML": str(
+                            Path(sim_struct.get_fault_VM_dir(root_folder, run_name))
+                            / "vm_params.yaml"
+                        ),
+                        "STAT_FILE": str(params.stat_file),
+                        "FAULT_DIR": fault_dir,
+                        "FDSTATLIST": str(Path(fault_dir) / f"fd{str(params.sufx)}.ll"),
+                        "MGMT_DB_LOC": root_folder,
+                        "REL_NAME": run_name,
```

Comment:
since the same key-string is repeating multiple times, and for the unlikely senario that the value/string of the keys are changed, would it be a good idea to have a mapping somewhere?

Reply:
after reading a bit more i realized that these keys really dont serve any purpose currently. the keys are not used in the scheduler at all.